File: server/services/storage.js
import fs from 'fs/promises';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const LINKS_FILE = path.join(__dirname, '../data/account-links.json');

export async function readLinks() {
    try {
        const data = await fs.readFile(LINKS_FILE, 'utf8');
        return JSON.parse(data);
    } catch (error) {
        if (error.code === 'ENOENT') {
            // File doesn't exist, create it with empty links array
            await fs.writeFile(LINKS_FILE, JSON.stringify({ links: [] }));
            return { links: [] };
        }
        throw error;
    }
}

export async function writeLinks(data) {
    await fs.writeFile(LINKS_FILE, JSON.stringify(data, null, 2));
}

File: server/src/services/storage.py
import json
import os
from pathlib import Path
from dotenv import load_dotenv

project_dir = Path(__file__).parents[2]
load_dotenv()

DATA_DIR = os.environ.get("DATA_DIR", "data")
LINKS_FILE = Path(project_dir / DATA_DIR / "account-links.json")


def read_links():
    try:
        if not LINKS_FILE.exists():
            LINKS_FILE.parent.mkdir(parents=True, exist_ok=True)
            write_links({"links": []})
            return {"links": []}

        with open(LINKS_FILE, "r") as f:
            return json.load(f)
    except Exception as e:
        raise Exception(f"Error reading links: {str(e)}")


def write_links(data):
    try:
        LINKS_FILE.parent.mkdir(parents=True, exist_ok=True)
        with open(LINKS_FILE, "w") as f:
            json.dump(data, f, indent=2)
    except Exception as e:
        raise Exception(f"Error writing links: {str(e)}")
